src/peakrdl_sv/node.py

```diff
@@ -109,9 +109,10 @@ def get_reg2hw_struct_bits(self) -> int:
         in the reg2hw struct that contains the 'q', 'qe', and 're' fields.
 
         """
-        if not self.is_hw_readable:
-            return 0
-        return self.width + int(self.needs_qe) + int(self.needs_qre)
+        bits = 0
+        if self.implements_storage:
+            bits += self.width
```

Comment:
```
bits = self.width if self.implements_storage else 0
```
 might me a bit more 'pythonic'. Purely a style thing so no need to change if you are unbothered.

Reply:
Fixed.